### PR TITLE
feat: split liquidity auction trigger into two cases

### LINKED ISSUE #7802: Split triggers for liquidity auction
# Feature Overview

When we are in a liquidity auction we have trigger == AUCTION_TRIGGER_LIQUIDITY but there are two possible triggers: 
1. not enough committed liquidity
2. can't deploy LP orders because there's nothing to peg on one or both sides of the book

Split the AUCTION_TRIGGER_LIQUIDITY into the two cases: AUCTION_TRIGGER_LIQUIDITY_TARGET_NOT_MET and AUCTION_TRIGGER_UNABLE_TO_DEPLOY_LP_ORDERS

### PR DESCRIPTION
**Pull request by ze97286**
Closes #7802

### CODE REVIEW

**Review by witgaw — DISMISSED** [review]
LGTM

File: core/monitor/auction.go [hunk 1]
```diff
@@ -63,9 +63,9 @@ func NewAuctionState(mkt *types.Market, now time.Time) *AuctionState {
 
 // StartLiquidityAuction - set the state to start a liquidity triggered auction
 // @TODO these functions will be removed once the types are in proto.
-func (a *AuctionState) StartLiquidityAuction(t time.Time, d *types.AuctionDuration) {
+func (a *AuctionState) StartLiquidityAuction(t time.Time, d *types.AuctionDuration, tigger types.AuctionTrigger) {
```
Comment (wwestgarth) [inline core/monitor/auction.go:66]: Passing the trigger type in fits badly with the other functions. I can currently call it like this:
```
StartLiquidityAuction(types.AuctionTriggerPrice)
```
and it'll set the state to be a price auction, despite the function being called `StartLiquidityAuction()`.

We should either have one general function that takes in a trigger for all cases, or split `StartLiquidityAuction()` into two.

Comment (wwestgarth) [reply core/monitor/auction.go:66]: Also a typo: `tigger` -> `trigger`.

File: protos/sources/vega/vega.proto [hunk 2]
```diff
@@ -75,8 +75,12 @@ enum AuctionTrigger {
   AUCTION_TRIGGER_OPENING = 2;
   // Price monitoring trigger
   AUCTION_TRIGGER_PRICE = 3;
-  // Liquidity monitoring trigger
+  // Deprecated
```
[the remark this answers is not in the document]
Comment (ze97286) [reply protos/sources/vega/vega.proto:78]: the reason i kept it is to simplify migration even if just for testing.

File: datanode/gateway/graphql/schema.graphql [hunk 3]
```diff
@@ -3650,8 +3650,10 @@ enum AuctionTrigger {
   AUCTION_TRIGGER_OPENING
   "Price monitoring"
   AUCTION_TRIGGER_PRICE
-  "Liquidity monitoring"
-  AUCTION_TRIGGER_LIQUIDITY
+  "Liquidity monitoring due unmet target stake"
```
Comment (candida-d) [inline datanode/gateway/graphql/schema.graphql:3653]:
```suggestion
  "Liquidity monitoring due to unmet target stake"
```

File: protos/sources/vega/vega.proto [hunk 4]
```diff
@@ -75,8 +75,12 @@ enum AuctionTrigger {
...
   AUCTION_TRIGGER_LIQUIDITY = 4;
+  // Liquidity auction due to not enough comitted liquidity
```
Comment (candida-d) [inline protos/sources/vega/vega.proto:80]:
```suggestion
  // Liquidity auction due to not enough committed liquidity
```

**Review by candida-d — CHANGES_REQUESTED** [review]
typo fixes in descriptions

**Review by candida-d — APPROVED** [review]

**Review by witgaw — APPROVED** [review]

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**